sdk/core/AzureCore/Source/Pipeline/Transport/URLSessionTransport.swift

```diff
@@ -84,6 +84,28 @@ public class URLSessionTransport: HTTPTransportStage {
             pipelineRequest.logger.error("Invalid session.")
             return
         }
+        guard let clientRequestIdHeader = pipelineRequest.httpRequest.headers[.clientRequestId],
```

Comment:
We should do this at the very top, so that any failures further down can also set the task's `state` to `failed`. I noticed that we're not ever setting the state to either `failed` or `completed` during the execution of the task. We should do that any time before we call a callback.

Reply:
Without the client request ID we can't access the task, so we won't be able to set the task state to anything without this unless we attach the task to the pipeline process. I'm not opposed to that, though the intent was really just to have a way of marking request IDs as canceled in order to interrupt the pipeline.

Reply:
Yep, that makes sense. My comment was more along the lines of we want to get the task right away, so that any failure like the one above on L84 can also set the task state to failed.

As a separate comment, I suggested we should set the task state to failed or completed any time we call the callback with either a success or an error.

However it sounds like this may no longer be applicable if we go with the cancellation token approach.